src/app.py

```diff
@@ -0,0 +1,235 @@
+"""
+import json, os, sys, letslog
+from flask import Flask, abort, request, jsonify
+from flask_restful import reqparse, abort, Api, Resource
+from configparser import ConfigParser
+from pymongo import MongoClient
+from pymongo import errors as mongoerrors
+from bson.json_util import dumps, loads
+
+app = Flask(__name__)
+api = Api(app)
+env_config = ConfigParser()
+env_config.read(os.path.dirname(os.path.abspath(__file__))+'/../config/setting.config')
+mongo_config = env_config['MongoDB']
+
+def get_db():
+    """Get MongoDB username and password from the config file and return the desired database.
+
+    Raises:
+        pymongo.errors: possibly connection errors or conficuration errors
+    Returns:
+        The database object
+
+    """
+    username, password = mongo_config['Mongo_User'], mongo_config['Mongo_Password']
+    db_name = mongo_config['Mongo_DBName']
+    client = MongoClient('mongodb+srv://' + username + ':' + password+ mongo_config['Mongo_Postfix'])
+    return client.get_database(db_name)
+db = get_db()
+
+def abort_if_course_doesnt_exist(course_id, courses, year=None, quarter=None):
+    """Send error message if course does not exist.
+
+    Send corresponding error message if a requested course does not exist
+
+    Args:
+        course_id: the crn of the course requested
+        courses: the dictionary of courses loaded from db
+        year: the year of the requested quarter
+        quarter: the requested quarter
+    Returns:
+        None
+
+    """
+    if course_id not in courses:
+        if year and quarter:
+            abort(404, message="Not Found(404): Course {0} doesn't exist in quarter {1} {2}".format(course_id, year, quarter))
+        else:
+            abort(404, message="Not Found(404): Course {} doesn't exist".format(course_id))
+
+def abort_invalid_input(err_message):
+    """Send an error message 400.
+    """
+    abort(400, message=err_message)
+
+def abort_not_found(err_message):
+    """Send an error message 404.
+    """
+    abort(404, message=err_message)
+
+def get_quarter_collections(year, quarter, collection_type):
+    """Get the collection from db according to defined quarter and the type of collection requested.
+
+    Args:
+        year: the year of the requested quarter
+        quarter: the requested quarter
+        collection_type: str, either "courses" or "departments"
+    Returns:
+        The collection received from db
+
+    """
+    collection_name = str(year) + " " + quarter + " De Anza " + collection_type
+    coll = db.get_collection(collection_name)
+    if coll.count() == 0:
+        abort(404, message="Not Found(404): Database collection entry {} doesn't exist".format(collection_name))
+    return coll
+
+
+
+parser = reqparse.RequestParser()
+parser.add_argument('year')
+parser.add_argument('quarter')
+parser.add_argument('department')
+parser.add_argument('course_id')
+
+class course(Resource):
```

Comment:
nit:
would it be better to put course, course list, and department list into separate .py files? in this case, if we want to add more and more functions to these classes, we can still keep the main file clean.